packages/terraform/src/utils/create-executor.ts

```diff
@@ -19,15 +21,33 @@ export function createExecutor(command: string) {
     }
 
     const { sourceRoot } = context.workspace.projects[context.projectName]
-    const { backendConfig = [] } = options
+    const { backendConfig = [],planFile, ciMode, autoApproval } = options
 
+    let plan = ''
+    let env = {}
+    
+    if( command === 'apply') {
+      plan = planFile
+    }
+    if(command === 'plan'){
+       plan = `-out ${planFile}`
+    }
+    if ( ciMode ) {
+       env = {
+        "TF_IN_AUTOMATION": true,
+        "TF_INPUT": 0,
+      }
+    }
     execSync(buildCommand([
       'terraform',
       command,
-      ...backendConfig.map((config) => `-backend-config="${config.key}=${config.name}"`)
+      ...backendConfig.map((config) => `-backend-config="${config.key}=${config.name}"`),
+      plan,
+      autoApproval ? "-auto-approve": ''
     ]), {
       cwd: sourceRoot,
-      stdio: 'inherit'
+      stdio: 'inherit',
+      env: {...process.env, ...env}
```

Comment:
`process.env` is not needed I think.

Reply:
I just want to be sure it take the existing one, you think it's not mandatory ?

Reply:
Yea you can keep it as is, see I do the same on other places. Better to be sure indeed :)